link_resolver_track_list.go

```diff
@@ -0,0 +1,205 @@
+		ParsedLink  string    `json:"parsedLink"`
+		Tags        []string  `json:"tags"`
+		Authors     []struct {
+			ID   int    `json:"ID"`
+			Name string `json:"name"`
+			Role string `json:"role"`
+		} `json:"authors"`
+	}
+
+	type TrackListAPIResponse struct {
+		Data TrackData `json:"data"`
+	}
+
+	tmpl, err := template.New("trackListEntryTooltip").Parse(tooltipTemplate)
+	if err != nil {
+		log.Println("Error initialization track list entry tooltip template:", err)
+		return
+	}
+
+	load := func(rawTrackID string, r *http.Request) (interface{}, error, time.Duration) {
+		trackID, _ := strconv.ParseInt(rawTrackID, 10, 32)
+		apiURL := fmt.Sprintf(trackListAPIURL, trackID)
+
+		// Create Track list API request
+		req, err := http.NewRequest("GET", apiURL, nil)
+		if err != nil {
+			return &LinkResolverResponse{
+				Status:  http.StatusInternalServerError,
+				Message: "Track list request creation error " + clean(err.Error()),
+			}, nil, noSpecialDur
+		}
+		req.Header.Set("User-Agent", "chatterino-api-cache/1.0 link-resolver")
+
+		// Execute Track list API request
+		resp, err := httpClient.Do(req)
+		if err != nil {
+			return &LinkResolverResponse{
+				Status:  http.StatusInternalServerError,
+				Message: "Track list http request error " + clean(err.Error()),
+			}, nil, noSpecialDur
+		}
+		defer resp.Body.Close()
+
+		// Error out if the track isn't found or something else went wrong with the request
+		if resp.StatusCode < http.StatusOK || resp.StatusCode > http.StatusMultipleChoices {
+			return trackNotFoundResponse, nil, noSpecialDur
+		}
+
+		// Read response into a string
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return &LinkResolverResponse{
+				Status:  http.StatusInternalServerError,
+				Message: "Track list http body read error " + clean(err.Error()),
+			}, nil, noSpecialDur
+		}
+
+		// Parse response into a predefined JSON blob (see TrackListAPIResponse struct above)
+		var jsonResponse TrackListAPIResponse
+		if err := json.Unmarshal(body, &jsonResponse); err != nil {
+			return &LinkResolverResponse{
+				Status:  http.StatusInternalServerError,
+				Message: "Track list api unmarshal error " + clean(err.Error()),
+			}, nil, noSpecialDur
+		}
+		if jsonResponse.Data.ID == 0 { // API responds with
+			return &LinkResolverResponse{
+				Status:  http.StatusNotFound,
+				Message: fmt.Sprintf("A track with ID %d doesn't exist.", trackID),
+			}, nil, noSpecialDur
+		}
+
+		trackData := jsonResponse.Data
+
+		prettyAuthors := ""
+		for i, elem := range trackData.Authors {
+			if i != 0 {
+				prettyAuthors += ", "
+			}
+			prettyAuthors += fmt.Sprintf("%s (%d; %s)", elem.Name, elem.ID, elem.Role)
+
+		}
+
+		// Build tooltip data from the API response
+		data := TooltipData{
+			ID:         trackData.ID,
+			Name:       trackData.Name,
+			AuthorName: prettyAuthors,
+			Tags:       strings.Join(trackData.Tags, ", "),
+			Duration:   time.Duration(trackData.Duration) * time.Second,
```

Comment:
This can be handled [just like with youtube videos](https://github.com/Chatterino/api/blob/master/link_resolver_youtube.go#L97)

Reply:
Fixed in `b73baa7`